### ISSUE TITLE
feat: @koi/registry-nexus — agent discovery & registration

### ISSUE THREAD

**Issue by windoliver** [windoliver/koi#193]
## Summary

Enable Koi agents to discover and resolve other running agents via Nexus's AGENT.json card system. Before Agent A can message Agent B, it needs to FIND Agent B — by skill, by name, or by ID.

**Blocks:** #192 (IPC is useless without discovery)

### Add `AgentRegistry` interface + `AgentDescriptor` type

```typescript
// New file: registry.ts — or add to ecs.ts

interface AgentRegistry {
  readonly discover: (query?: string) => Promise<readonly AgentDescriptor[]>;
  readonly resolve: (agentId: string) => Promise<AgentDescriptor | undefined>;
}

interface AgentDescriptor {
  readonly agentId: string;
  readonly name: string;
  readonly skills: readonly string[];
  readonly status: string;
}
```

**Design rationale — why this is slim:**
- `discover(query?)` — string query, backend decides how to interpret (keyword, skill match, regex)
- `resolve(id)` — point lookup by ID
- No `register()`/`deregister()` in L0 — registration is a WRITE concern, handled by L2 ComponentProviders
- `AgentDescriptor` has 4 fields — minimal card info. Nexus has more (metadata, inbox path, created_at) but those are L2 concerns
- `status` is `string` not an enum — L2 decides the vocabulary ("active", "idle", "suspended", etc.)

### What it implements

| L0 Interface | Nexus Backend |
|---|---|
| `AgentRegistry.discover()` | `AgentDiscovery.discover_all()` — scans `/agents/`, reads AGENT.json cards |
| `AgentRegistry.discover("skill:search")` | `AgentDiscovery.find_by_skill("search")` — filtered scan |
| `AgentRegistry.resolve(id)` | `AgentDiscovery.get_agent_card(id)` — reads `/agents/{id}/AGENT.json` |
| `ComponentProvider.attach()` | Calls `AgentProvisioner.provision()` + writes AGENT.json from manifest |
| `ComponentProvider.detach()` | Updates AGENT.json status to "terminated" |

### Manifest → AGENT.json bridge

```
koi.yaml (Koi manifest)          AGENT.json (Nexus card)
─────────────────────────         ─────────────────────────
name: "Research Bot"        →     name: "Research Bot"
tools: [search, browse]     →     skills: ["search", "browse"]
(runtime state)             →     status: "active"
(computed path)             →     inbox: "/agents/research-bot/inbox"
```

### Configuration

```yaml
# koi.yaml
registry:
  backend: "@koi/registry-nexus"
  options:
    nexusUrl: "http://localhost:8000"
    zone: "default"
    cacheTtlMs: 10000
```

### Alternative L2 backends

- `@koi/registry-memory` — in-memory Map (tests)
- `@koi/registry-gateway` — Gateway in-memory registry (single-node)

## Acceptance Criteria

- [ ] `AgentRegistry` and `AgentDescriptor` types added to `@koi/core`
- [ ] `@koi/registry-nexus` package created
- [ ] Agent registration on `ComponentProvider.attach()` writes AGENT.json
- [ ] `discover()` returns all active agents
- [ ] `discover("skill:search")` filters by skill
- [ ] `resolve(id)` returns single agent or undefined
- [ ] Status updated on state transitions (via EventComponent listener)
- [ ] Status updated to "terminated" on `detach()`
- [ ] Cached discovery with configurable TTL
- [ ] In-memory test backend
- [ ] 80%+ test coverage

## References

- Nexus discovery: `~/nexus/src/nexus/ipc/discovery.py`
- Nexus provisioning: `~/nexus/src/nexus/ipc/provisioning.py`
- Nexus agent card: `~/nexus/src/nexus/bricks/a2a/agent_card.py`
- A2A spec: `GET /.well-known/agent.json`
